src/metric/regression_metric.hpp

```diff
@@ -162,5 +162,23 @@ class FairLossMetric: public RegressionMetric<FairLossMetric> {
   }
 };
 
+/*! \brief Poisson regression loss for regression task */
+class PoissonMetric: public RegressionMetric<PoissonMetric> {
+public:
+  explicit PoissonMetric(const MetricConfig& config) :RegressionMetric<PoissonMetric>(config) {
+  }
+
+  inline static double LossOnPoint(float label, double score, double, double) {
+    const double eps = 1e-10f;
+    if (score < eps) {
+      score = eps;
```

Comment:
Ah yes that's true. Myself I was using eps in one of my packages for Poisson regression...